anker/src/state.rs

```diff
@@ -235,6 +272,102 @@ impl Anker {
     ) -> ProgramResult {
         Anker::check_is_spl_token_account("Solido's stSOL", &solido.st_sol_mint, token_account_info)
     }
+
+    /// Check the if the token swap program is the same as the one stored in the
+    /// instance.
+    ///
+    /// Check all the token swap associated accounts.
+    /// Check if the rewards destination is the same as the one stored in Anker.
+    pub fn check_token_swap(
+        &self,
+        anker_program_id: &Pubkey,
+        accounts: &SellRewardsAccountsInfo,
+    ) -> ProgramResult {
+        // Check token swap instance parameters.
+        if &self.pool != accounts.pool.key {
+            msg!(
+                "Invalid Token Swap instance, expected {}, found {}",
+                self.pool,
+                accounts.pool.key
+            );
+            return Err(AnkerError::WrongSplTokenSwap.into());
+        }
+        // We should ignore the 1st byte for the unpack.
```

Comment:
I think I commented on this before, did I miss the reply? Why do we need to skip the first byte, and shouldn’t we validate the value?

Reply:
Done! ^^ The value should be 1